lib/dependencies/all.js

```diff
@@ -0,0 +1,75 @@
+goog.provide('shaka.dependencies');
+
+/**
+ * @export
+ */
+shaka.dependencies = class {
+  /**
+   * Registers a new dependency.
+   *
+   * @param {shaka.dependencies.Allowed} key which is used for retrieving a
+   * dependency
+   * @param {?} dep a dependency
+   * @export
+   */
+  static add(key, dep) {
+    if (!shaka.dependencies.Allowed[key]) {
+      throw new Error(`${key} is not supported`);
+    }
+    shaka.dependencies.dependencies_[key] = dep;
+  }
+
+  /**
+   * Retrieve dependency by provided key.
+   *
+   * @param {shaka.dependencies.Allowed} key key of a dependency
+   * @return {?} any dependency
+   * @export
+   */
+  static get(key) {
+    return shaka.dependencies.dependencies_[key];
+  }
+
+  /**
+   * Check if we have a dependency fro the key
+   * @export
+   * @param {string} key key
+   * @return {boolean}
+   */
+  static has(key) {
+    return !!shaka.dependencies.dependencies_[key];
+  }
+
+  /**
+   * @return {muxjs}
```

Comment:
It doesn't work for me. Without type casting, I got an error: "ERROR - [JSC_TYPE_MISMATCH] inconsistent return type"
`
/** @return {typeof muxjs} */
  static muxjs() {
    return shaka.dependencies.dependencies_[
        shaka.dependencies.Allowed.muxjs];
  }
`
but the construction 
`(/** @type {typeof muxjs} */(shaka.dependencies.dependencies_[
        shaka.dependencies.Allowed.muxjs]
    ));` is illegal, since I got the error message: "ERROR: Unable to parse a tag's type expression for source file ... in line 32 with tag title "return" and text "{typeof muxjs}": Invalid type expression "typeof muxjs": Expected "|" but "m" found."

Reply:
@TheModMaker if you still believe this is important, can you help me make it work?